### PR TITLE
Improve flake8 output in travis-ci (#452)

### PR DESCRIPTION
**Pull request by reszelaz**
This PR aims to improve the output of the flake8 check in travis-ci - #452.

### CODE REVIEW

**Comment by cpascual** [comment]
The commit c592f92 seems ok to me.
Is this still WIP? (the title says so)

Are you planing to also improve the output as per https://github.com/sardana-org/sardana/issues/452#issuecomment-302637286 ?

**Comment by reszelaz** [comment]
Yes, I will either improve the output or the documentation.
I will try to have it for tomorrow.

**Comment by reszelaz** [comment]
Now it is ready for review and eventual integration.

**Comment by cpascual** [comment]
With the latest commit (d478876) you introduced a bug: the script returns 0 even if flake8 found errors

**Comment by cpascual** [comment]
I am fixing it myself and merging

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,12 @@ services:
 python:
   - "2.7"
 
+env:
+  # run flake8 check on all python files in the project
+  - TEST_CMD="ci/flake8_diff.sh"
+  # run the full testsuite
+  - TEST_CMD="docker exec sardana-test sardanatestsuite"
+
```
Comment (cpascual) [inline .travis.yml]: I just realised that this implementation makes the flake8 test also run a lot of the `install` part that is useless for it. This has 2 problems:
- It wastes time (the flake8 test could be much faster)
- It mixes cases for failure: e.g., if a commit introduces problems that make  `python setup.py install` to fail, the flake8 test would report an error that is actually due to the unrelated install issue

To fix this, we could use [stages](https://docs.travis-ci.com/user/build-stages) but maybe it is simpler to just do the install steps conditionally, as in the following example::

```yaml
(...)
env:
  - TEST_CMD="ci/flake8_diff.sh" INSTALL="N"
  - TEST_CMD="docker exec sardana-test sardanatestsuite"  INSTALL="Y"

(...)

install:
  - if [ $INSTALL == "Y" ]; then docker pull reszelaz/sardana-test ; fi
  - if [ $INSTALL == "Y" ]; then docker run -d --name=sardana-test -h sardana-test --volume=`pwd`:/sardana reszelaz/sardana-test ; fi
  - (...)

(...)


```

Comment (reszelaz) [inline .travis.yml]: Using stages would mean changing the approach completely,  since they run sequentially and  a stage failure makes the whole pipe to break. That would mean loosing information in case of a failure - we will only learn about the state of one of the checks.

Apart of the option that you suggest (conditional install) we could also revert just the commit c592f92, accepting the two others and run one job only. 

I will investigate a little bit more on that..

Comment (cpascual) [inline .travis.yml]: > we could also revert just the commit c592f92, accepting the two others and run one job only.

IMHO that would be a pity. I would go for conditional install

**Comment by reszelaz** [comment]
I have added conditional tests preparations and it improved the jobs durations. Now the `flake8` takes only 37 s.

The travis configuration file may be hard to follow now. In order to improve it, we could move the whole `testsuite` part to a script (both `install` and `script` sections commands), let's call it `testsuite_docker.sh`. With that, we could just run `falke8_diff.sh` or `testsuite_docker.sh` in the `script` section depending on the env.
 
This way we will gain some readability in the travis configuration file but in the jobs output all theses commands, that now are separate, will be treated as one. Now each of the commands of the `install` section is by default folded, so its output is hidden until someone unfold it. Moving it to a script we will loose this making the output less readable. 

At the end, I think that the less readable configuration file is less harmful than the less readable jobs output - the one that the users read. So I would leave it as it is. But I would appreciate your opinions.
